tools/src/test/java/org/apache/kafka/tools/consumer/group/ResetConsumerGroupOffsetTest.java

```diff
@@ -415,153 +474,246 @@ public void testResetOffsetsExportImportPlan() throws Exception {
-        ConsumerGroupCommand.ConsumerGroupService consumerGroupCommandExec2 = getConsumerGroupService(cgcArgsExec2);
-        Map<String, Map<TopicPartition, OffsetAndMetadata>> importedOffsets2 = consumerGroupCommandExec2.resetOffsets();
-        assertEquals(exp1, toOffsetMap(importedOffsets2.get(group1)));
+            Map<String, Map<TopicPartition, OffsetAndMetadata>> importedOffsets2 = serviceExec2.resetOffsets();
+            assertEquals(exp1, toOffsetMap(importedOffsets2.get(group1)));
 
-        adminZkClient().deleteTopic(TOPIC);
+            admin.deleteTopics(asList(topic1, topic2));
+        }
     }
 
-    @Test
-    public void testResetWithUnrecognizedNewConsumerOption() {
-        String[] cgcArgs = new String[]{"--new-consumer", "--bootstrap-server", bootstrapServers(listenerName()), "--reset-offsets",
-            "--group", GROUP, "--all-topics", "--to-offset", "2", "--export"};
+    @ClusterTemplate("generator")
+    public void testResetWithUnrecognizedNewConsumerOption(ClusterInstance cluster) {
+        String[] cgcArgs = new String[]{"--new-consumer",
+            "--bootstrap-server", cluster.bootstrapServers(),
+            "--reset-offsets", "--group", GROUP, "--all-topics",
+            "--to-offset", "2", "--export"};
         assertThrows(OptionException.class, () -> getConsumerGroupService(cgcArgs));
     }
 
-    private void produceMessages(String topic, int numMessages) {
-        List<ProducerRecord<byte[], byte[]>> records = IntStream.range(0, numMessages)
-            .mapToObj(i -> new ProducerRecord<byte[], byte[]>(topic, new byte[100 * 1000]))
-            .collect(Collectors.toList());
-        kafka.utils.TestUtils.produceMessages(servers(), seq(records), 1);
+    private Map<TopicPartition, Long> committedOffsets(ClusterInstance cluster, String topic, String group) {
+        try (Consumer<String, String> consumer = createNoAutoCommitConsumer(cluster, group)) {
+            Set<TopicPartition> partitions = consumer.partitionsFor(topic)
+                    .stream()
+                    .map(partitionInfo -> new TopicPartition(partitionInfo.topic(), partitionInfo.partition()))
+                    .collect(Collectors.toSet());
+            return consumer.committed(partitions)
+                    .entrySet()
+                    .stream()
+                    .filter(e -> e.getValue() != null)
+                    .collect(toMap(Map.Entry::getKey, e -> e.getValue().offset()));
+        }
     }
 
-    private void produceConsumeAndShutdown(String topic, String group, int totalMessages, int numConsumers) throws Exception {
-        produceMessages(topic, totalMessages);
-        ConsumerGroupExecutor executor = addConsumerGroupExecutor(numConsumers, topic, group, GroupProtocol.CLASSIC.name);
-        awaitConsumerProgress(topic, group, totalMessages);
-        executor.shutdown();
+    private Consumer<String, String> createNoAutoCommitConsumer(ClusterInstance cluster, String group) {
```

Comment:
Could you test the new consumer also? 

https://github.com/apache/kafka/blob/trunk/core/src/test/java/kafka/test/ClusterInstance.java#L157

`ClusterInstance` can return the supported protocol, and you can leverage it to create specify consumer.